lighthouse-extension/app/src/lighthouse-background.js

```diff
@@ -124,19 +147,30 @@ window.runLighthouseForConnection = function(connection, url, options, requested
   lighthouseIsRunning = true;
   updateBadgeUI(url);
 
-  // Run Lighthouse.
-  return Runner.run(connection, runOptions)
-    .then(result => {
-      lighthouseIsRunning = false;
-      updateBadgeUI();
-      filterOutArtifacts(result);
-      return result;
-    })
-    .catch(err => {
-      lighthouseIsRunning = false;
-      updateBadgeUI();
-      throw err;
-    });
+  return enableOtherChromeExtensions(false)
+      .then(_ => Runner.run(connection, runOptions)) // Run Lighthouse.
+      .then(result => {
+        lighthouseIsRunning = false;
+        updateBadgeUI();
+        filterOutArtifacts(result);
+
+        // Deliver results even if enabling extensions fails.
+        enableOtherChromeExtensions(true).catch(err => {
+          log.warn('Chrome', `Could not disable some extensions. ${err.message}`);
+        });
+
+        return result;
+      })
+      .catch(err => {
+        lighthouseIsRunning = false;
+        updateBadgeUI();
+
+        enableOtherChromeExtensions(true).catch(err => {
+          log.warn('Chrome', `Could not enable/disable some extensions. ${err.message}`);
```

Comment:
good question. I don't actually know a great way, but I'm concerned there will be some corner case of permissions or something where we end up silently disabling some people's extensions and we won't know about it until someone notices (or we get error reporting integrated)